File: programming_assignment_1/gcd.rb
def gcd(a, b)
  while b > 0 do
    remainder = a % b
    a = b
    b = remainder
  end
  a
end

if __FILE__ == $0
  a, b = gets.split().map(&:to_i)
  puts "#{gcd(a, b)}"
end

File: programming_assignment_1/gcd.py
import sys

def gcd_naive(a, b):
    
    while b > 0:
        remainder = a % b
        a = b
        b = remainder
        
    return a

if __name__ == "__main__":
    input = sys.stdin.read()
    a, b = map(int, input.split())
    print(gcd_naive(a, b))
